services/indexes/avax/reader_list_cblocks.go

```diff
@@ -116,9 +117,10 @@ func (r *Reader) ListCBlocks(ctx context.Context, p *params.ListCBlocksParams) (
 			"status",
 			"gas_used",
 			"gas_price",
+			"block_idx",
```

Comment:
What is the purpose of adding block_idx in the result set?

Reply:
The query transformation via the code is getting the same result with another way. The end query is for example:

SELECT DISTINCT serialization, created_at, F.address AS from_addr, block, idx, status, gas_used, gas_price, block_idx 
FROM (select id from cvm_accounts where address in ('0x70110644eb7c13b2044b650cff84421463941c61') ) sub,cvm_transactions_txdata 
LEFT JOIN `cvm_accounts` AS `F` ON id_from_addr = F.id WHERE (block_idx IS NOT NULL) AND ((id_from_addr=sub.id  OR id_to_addr=sub.id)) 
ORDER BY block_idx DESC LIMIT 50

The block_idx was added in order to maintain the ORDER BY.

And why this was done this way? Because we needed to get the unique records via distinct clause because with this structure the  **(select id from cvm_accounts where address in ('0x70110644eb7c13b2044b650cff84421463941c61') ) sub** would be consider the first table of truth since we are having the ( condition OR condition ) within the where clauses.

Without the distinct clause we could have duplicate records that would actually be valid for the ( condition OR condition ) which is actually the ((id_from_addr=sub.id  OR id_to_addr=sub.id))

Reply:
Understand, thx!